### PR TITLE
feat: enable socialgouv autodevops

### PR DESCRIPTION
**Pull request by revolunet**
enable feature branches on gitlab

to check :

 - removed knex production seeds
 - brcypt password in development seeds

### CODE REVIEW

File: .k8s/app.values.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      kubernetes.io/ingress.class: "azure/application-gateway"
+   hosts:
+      - host: "medle.dev.fabrique.social.gouv.fr"
+        paths:
+           - path: /
+             servicePort: 3000
+   tls:
+      - hosts:
+           - "medle.dev.fabrique.social.gouv.fr"
+        secretName: wildcard-crt
```
Comment (douglasduteil) [inline .k8s/app.values.yml]: The `.k8s/app.values.yml` should contain production like values.
The certificate will be force to "wildcard-crt" if PRODUCTION env var in set (in dev and preprod).

```suggestion
        secretName: medle-crt
```

**Comment by revolunet** [comment]
should the hostname in app.values.yml be the production one ? 

i'll keep `.dev` at the moment while prod is still a manual deploy (vm)

File: Dockerfile [hunk 2]
```diff
@@ -31,10 +31,13 @@ ENV JWT_SECRET=$JWT_SECRET
...
 ENV TEST_CURRENT_DATE=$TEST_CURRENT_DATE
 
+ARG API_URL
+ENV API_URL=$API_URL
+
 ENV NODE_ENV=production
 ENV NEXT_TELEMETRY_DISABLED=1
 
-RUN yarn build
+RUN yarn build && yarn --production
```
Comment (douglasduteil) [inline Dockerfile]: might be better if the build is done outside of the docker build so the image is prod only

**Comment by revolunet** [comment]
@douglasduteil tu vois des parties a extraire dans gitlab-ci-yml?

**Comment by revolunet** [comment]
@pom cette pr te permet d'avoir des environnements déployés via des PR

Le merge to master envoie sur le cluster de prod sur une adresse temporaire pour linstant

File: Dockerfile (hunk 2 above)
Comment (revolunet) [reply Dockerfile]: you mean 2-stage build ?

**Comment by revolunet** [comment]
we mant to wait for https://github.com/SocialGouv/gitlab-ci-yml/pull/69 to be finished

**Comment by revolunet** [comment]
any idea to simplify the port stuff in `app.values.yml` ?

**Comment by revolunet** [comment]
do not merge yet plz

File: Dockerfile (hunk 2 above)
Comment (douglasduteil) [reply Dockerfile]: I mean we do on gitlab jobs we yarn and yarn build then we export the result of the build in artefacts to be used in the docker build where we "yarn --production" and copy the build artefacts.

File: package.json [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
-    "dotenv": "^8.2.0",
-    "env-cmd": "^10.1.0",
-    "eslint": "^6.5.1",
-    "file-loader": "^4.2.0",
-    "jest-cli": "^24.9.0",
-    "prettier": "^1.18.2",
-    "prop-types": "^15.7.2",
-    "url-loader": "^2.2.0"
-  }
+   "name": "medle",
```
Comment (douglasduteil) [inline package.json]: Strange indentation here :warning:

**Comment by revolunet** [comment]
replaced by #44

**Pull request closed without merging**